### PR TITLE
MOON-198: Adds ability to search in Menu

### PR DESCRIPTION
**Pull request by vindhya**
## JIRA

https://jira.jahia.org/browse/MOON-198

## Description

* Adds ability search in Menu
* Added documentation for search variant of Input

## Checklist

- [x] All files present (component - md - scss - spec - stories)
- [x] All props ok and documented (propTypes or typescript types)
- [x] Required props, default values, variants and colors
- [x] Example in storybook

## Documentation

- [x] README documentation (specification of the component, responsiveness, dependencies on other components, states, behaviours, animations, accessibility)

### CODE REVIEW

File: src/components/Menu/Menu.stories.jsx [hunk 1]
```diff
@@ -187,4 +187,111 @@ storiesOf('Components|Menu', module)
                 />
             </Menu>
         </div>
+    ))
+    .add('With Search', () => (
```
Comment (ffffffelix) [inline src/components/Menu/Menu.stories.jsx:191]: When I have the addon panel the menus overlap each over. As it is the same behavior for all variants, I think we can have just one menu type to present the search behavior.

**Review by ffffffelix — APPROVED** [review]

File: src/components/Menu/Menu.tsx [hunk 2]
```diff
@@ -215,6 +269,8 @@ export const Menu: React.FC<IMenuProps> = (
 Menu.defaultProps = {
     variant: MenuVariant.DEFAULT,
     hasOverlay: true,
+    hasSearch: false,
+    searchEmptyText: '',
```
[the remark this answers is not in the document]
Comment (vindhya) [reply src/components/Menu/Menu.tsx:273]: I was considering this but then we'd be forcing it into one language. If you still think providing default text here is preferable, I can add what is shown in the figma design.

File: src/components/Menu/Menu.stories.jsx (hunk 1 above)
Comment (vindhya) [reply src/components/Menu/Menu.stories.jsx:191]: Sounds good! I was mostly using it for testing when developing - I've removed the extra ones now and only left one.

File: src/components/Menu/Menu.tsx [hunk 3]
```diff
@@ -196,13 +235,28 @@ export const Menu: React.FC<IMenuProps> = (
...
+                            onChange={e => setInputValue(e.target.value)}
+                            onClear={() => setInputValue('')}
+                        />
+                    </div>
+                )}
+                {filteredChildren}
+                {isEmptySearch && (
+                    <div className="moonstone-menu_emptySearchText">
+                        <Typography variant="caption">{searchEmptyText}</Typography>
+                    </div>
```
[the remark this answers is not in the document]
Comment (vindhya) [reply src/components/Menu/Menu.tsx:252]: Good point! Changed.

**Review by dgriffon — APPROVED** [review]

File: src/components/Menu/Menu.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ffffffelix) [reply src/components/Menu/Menu.tsx:276]: I think is better to provide a default text even if it’s in English to show the best practice.
We can simply display: `No result found`.

[the remark this answers is not in the document]
Comment (vindhya) [reply src/components/Menu/Menu.tsx:276]: Ok! I'll make the change 🙂

**Pull request merged**